src/components/SlateEditor/plugins/PluginFactory.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import process from "process";
+import { Editor, NodeEntry, Element, Node, NodeInterface } from "slate";
+import { NormalizerConfig, defaultBlockNormalizer } from "../utils/defaultNormalizer";
+
+interface Props<T extends Node = Node> {
+  type: Element["type"];
+  normalize?: Normalize<T>[];
+  normalizerConfig?: NormalizerConfig;
+  onKeyDown?: Record<string, KeyDown>;
+  isInline?: boolean;
+  isVoid?: boolean;
+}
+
+interface Normalize<T extends Node = Node> {
+  description: string;
+  normalize(e: NodeEntry<T>, editor: Editor): boolean;
+}
+
+type KeyDown = (e: KeyboardEvent, editor: Editor, nextOnKeyDown?: (event: KeyboardEvent) => void) => void;
+
+const createPluginFactory =
+  <T extends Node = Node>({ normalize, isVoid, type, onKeyDown, isInline, normalizerConfig }: Props<T>) =>
+  (editor: Editor) => {
+    const {
+      isVoid: nextIsVoid,
+      normalizeNode: nextNormalizeNode,
+      isInline: nextIsInline,
+      onKeyDown: nextOnKeyDown,
+    } = editor;
+
+    editor.isVoid = (e) => (e.type === type ? !!isVoid : nextIsVoid?.(e));
+    editor.isInline = (e) => (e.type === type ? !!isInline : nextIsInline?.(e));
+
+    editor.normalizeNode = (entry) => {
+      const [node] = entry;
+      if (Element.isElement(node) && node.type === type) {
```

Comment:
Mistenker at dette ikke kommer til å holde. Når vi sjekker etter eksterne embeds sjekker vi for eksempel etter både `external` og `iframe`

Reply:
Hmm, det er jo mega kjipt.
Men ser at de er mega minimalistiske, kan vi ikke bare evt. lage plugins for de?

Reply:
Nei, egentlig ikke. Man vet ikke om noe er en `external` eller `iframe` før man prøver å lagre de.

Reply:
Det kommer ikke til å bli et problem. Vi kan bare lage External plugin med en child plugin som dekker casene for enten iframe eller external. Bing smack.